src/scipp/core/binning.py

```diff
@@ -0,0 +1,99 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+# @author Simon Heybrock
+from typing import Dict, List, Optional, Union
+
+from .._scipp import core as _cpp
+from .variable import array, Variable, linspace, arange
+from .bins import bin, histogram
+
+
+def _get_coord(x, name):
+    event_coord = x.bins.meta.get(name) if x.bins is not None else None
+    return x.meta.get(name, event_coord)
+
+
+def _upper_bound(x):
+    import numpy as np
+    bound = x.max()
+    bound.value = np.nextafter(bound.value, np.inf)
+    return bound
+
+
+def _parse_coords_arg(x, name, arg):
+    coord = _get_coord(x, name)
+    if isinstance(arg, int):
+        coord = linspace(name, coord.min(), _upper_bound(coord), num=arg + 1)
+    elif isinstance(arg, Variable):
+        if arg.ndim == 0:
```

Comment:
This has a potentially confusing edge case: `arg = sc.index(100)` means 'bins of width 100' but looks a lot like `arg = 100` which means '100 bins'.

Reply:
True, but you would notice from the output, no hidden bugs, I suppose?

Reply:
If you inspect the output directly. People might call `bin`, etc. in the middle of some complicated function and never see the output, only a processed version of it.

That being said, I don't see an alternative apart from not allowing keyword arguments but specifically named args like `edges` and `n_bins`.

Reply:
Based on user feedback we decided that this general approach is good.